smart_simulation/database/db_utils.py

```diff
@@ -0,0 +1,84 @@
+import logging
+import pathlib
+import sqlite3
+
+import pandas as pd
+
+import daiquiri
+from smart_simulation.cfg_templates.config import package_dir
+
+daiquiri.setup(
+    level=logging.INFO,
+    outputs=(
+        daiquiri.output.Stream(
+            formatter=daiquiri.formatter.ColorFormatter(
+                fmt="%(asctime)s [%(levelname)s] %(name)s.%(" "funcName)s: %(message)s"
+            )
+        ),
+    ),
+)
+
+PACKAGE_PATH = pathlib.Path(package_dir)
+LOCAL_PROD_DB = PACKAGE_PATH / "smart_simulation" / "database" / "production.db"
+
+
+def connect(database: pathlib.PurePath = LOCAL_PROD_DB) -> sqlite3.connect:
+    """
+    Connect to a sqlite3 database and return the connection.
+    Args:
+        database: Database to connect to, defaulted to local production database.
+
+    Returns: The database connection
+    """
+    if not isinstance(database, pathlib.PurePath):
+        logging.exception("weights_directory must be a pathlib Path.")
+        raise TypeError
+    if database.suffix != ".db":
+        logging.exception(f"Database: {database}, must be of extension type '.db'.")
+        raise TypeError
+    try:
+        connection = sqlite3.connect(database)
+        return connection
+    except Exception:
+        raise
+
+
+def weight_csv_to_db(
+    csv_file: pathlib.Path, truncated_uuid: str, database: pathlib.Path = LOCAL_PROD_DB
+):
+    """
+    Add weight csv file to the weights table of a the local database.
+    Args:
+        csv_file: Simulation weight output file path.
+        truncated_uuid: 8 character long truncated uuid.
+        database: Database to write to, defaulted to the local production database.
+    """
+    # TODO add in check for datetime format of index, and that weight column exists
+    # TODO and create temp df with just date_time, scale_id, weight
+    connection = connect(database)
```

Comment:
Any time you see an `open`/`close` pattern in Python you should consider using a `with` statement / context manager. Here I think it should just amount to:
```
with connect(database) as c:
     ... do stuff
```
`c.close()` will be automatically called when it finishes

Reply:
Noted, will make the change.